### DISCUSSION TITLE
Keyboard Swipe in Sideband

### DISCUSSION THREAD

**Discussion by HWCL** [markqvist/Reticulum#291] [Sideband]
This is a question for Mark -- it looks like swipe and autocorrect are disabled on the google keyboard when using Sideband. Is this intentional? Would it be possible to enable that functionality in a future release of Sideband?

Thanks!

**Comment by markqvist** [comment]
Ok, that is weird, I have no idea why. I don't use the google keyboard myself, so had not noticed something like that. It should just use whatever input method you have set, and in fact I am mostly using it with [this completely odd input method that I wrote](https://unsigned.io/log/2023_01_30_Penboard_Resurrecting_Graffiti.html), which works fine.

It may be the underlying Kivy UI framework doing something weird to disable certain IME behaviour. I'll look into it when I get a moment, but if anyone else knows anything, please do chime in!

**Reply by faragher** [reply]
Just as a data point, I also don't have autocomplete on my keyboard. I assumed it was a conscious security decision, but if the intention was to use the default keyboard, it doesn't appear to do so. (GBoard input, Android 9)

**Reply by HWCL** [reply]
Found this discussion on StackOverflow. Sounds like it's got to do with a bug in SDL:

https://stackoverflow.com/questions/63166102/setting-google-keyboard-in-kivy-and-kivymd

**Reply by faragher** [reply]
Looks like they pushed a fix in SDL in November, but it's not part of any branch? Since buildozer doesn't seem to work as-is, maybe targeting the known working versions would be the easiest fix, if none of the newer features are needed.

**Reply by markqvist** [reply]
Yeah, it's a bit annoying. Lot's of bugfixes in both SDL and Kivy projects waiting to be all collected into the build recipes in p4a and buildozer, which are themselves currently broken due to Android SDK dependency issues :S I have a whole host of weird patches to currently get the app building. It is a truly tedious situation. It looks a bit like a situation of everyone waiting on everyone before things can be resolved. Hopefully the next releases will be out soon, it looks like it at least.
